packages/blade/src/components/Link/_decisions/decisions.md

```diff
@@ -0,0 +1,33 @@
+# Link Decisions <!-- omit in toc -->
+- [Design](#design)
+- [Link Component](#link-component)
+  - [Link API](#link-api)
+- [Component Breakdown](#component-breakdown)
+- [Open Questions](#open-questions)
+
+## Design
+[Figma Link](https://www.figma.com/file/jubmQL9Z8V7881ayUD95ps/Blade---Payment-Light?node-id=12699%3A147155) to the **Link** component
+
+
+## Link Component
+A Link component that can act as a `<button>` as well as an `<anchor>` element
+
+### Link API
+
+| Prop | Type | Required | Default | Description |
+|---|---|---|---|---|
+| variant | `anchor`, `button` | No | `anchor` | The variant of the Link to be rendered.<br><br>Note: `button` will look like an inline Link component(no spacing; can be rendered alongside other text) but will be rendered as a `button` element |
+| children | `string` | No | undefined | The text to be rendered within the link. |
+| icon | `Icon` | No | undefined | The Blade `Icon` component to be rendered within the link. |
+| iconPosition | `left`, `right` | No | `right` | The position of the rendered icon. |
+| isDisabled | `boolean` | No | `false` | Control whether the link is disabled or not. |
+| onClick | `Function` | No | undefined | The function to be called when the link is clicked. |
```

Comment:
+1. We might need to accept both because of scenarios like analytics libraries where user might need to send some event when the link is Clicked. 

> Should we consider scenarios where devs want the onClick to override the href but still want it as a link and in that case the href can be equated to javascript:void(0)?

If we devs want the onClick to override the href they can just `return false` from the onClick right?

Reply:
`href` should do what it natively does, take the user to a specific place. `onClick` should be called irrespective of `href`. Consumers would need this in cases where they might want to track click events with some analytics service after the link is clicked and that logic can go inside `onClick`.

Reply:
Since clicking with `href` can take people do a different page, do we handle waiting for the `onClick` handler to finish before the page unmounts?

Reply:
`href` is useful for navigation, SEO, keyboard press